src/lib/io/sol-i2c-common.c

```diff
@@ -138,10 +139,19 @@ i2c_dispatcher_exec_op(void *data)
         return true;
     }
 
-    if (!sol_i2c_set_slave_address(shared->i2c, op_set->addr)) {
-        SOL_ERR("Failed to set slave address 0x%02x on I2C bus: %d.", op_set->addr,
-            sol_i2c_bus_get(shared->i2c));
+    r = sol_i2c_set_slave_address(shared->i2c, op_set->addr);
+    if (r < 0) {
+        if (r == -EBUSY) {
+            if (++dispatcher->retry >= SOL_I2C_MAX_RETRIES) {
+                SOL_ERR("Failed to schedule I2C operation.");
+                goto exit;
+            }
+        }
+
+        SOL_ERR("Failed to set slave address 0x%02x on I2C bus: %d.",
+            op_set->addr, sol_i2c_bus_get(shared->i2c));
         i2c_dispatcher_end_set(dispatcher, -1);
+
```

Comment:
in case of `r < 0` it shouldn't return `true` only if `r == -EBUSY` and the number of retry is less than `SOL_I2C_MAX_RETRIES` ?